### PR TITLE
feat(quantic): new property fetchOnInit added to the Quantic Case Classification

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-5029](https://coveord.atlassian.net/browse/SFINT-5029)

Added a new property `fetchOnInit` to the Quantic Case Classifications  component, this property, when set to `true`, will trigger a request to fetch classifications during the initialization of the component.

## Demo from the Case Assist Cookbook:

https://github.com/coveo/ui-kit/assets/86681870/f55eb59b-7807-4ebf-9e1f-5c1d73e2434a

[SFINT-5029]: https://coveord.atlassian.net/browse/SFINT-5029?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/cypress/e2e/default-1/case-classification/case-classification.cypress.ts [hunk 1]
```diff
@@ -1180,9 +1181,34 @@ describe('quantic-case-classification', () => {
...
           );
-          Expect.fetchDocumentsAfterValueChange();
+          Expect.fetchDocumentSuggestions();
         });
       });
     });
   });
+
+  describe('when the fetchOnInit property is set to true', () => {
+    it('should automatically fetch classification during initialization', () => {
```
Comment (mmitiche) [inline packages/quantic/cypress/e2e/default-1/case-classification/case-classification.cypress.ts:1191]:
```suggestion
    it('should automatically fetch classifications during initialization', () => {
```

File: packages/quantic/cypress/e2e/default-1/case-classification/case-classification-expectations.ts [hunk 2]
```diff
@@ -164,13 +164,13 @@ function caseClassificationExpectations(selector: CaseClassificationSelector) {
         .logDetail('should log the "ticket_classification_click" UA event');
     },
 
-    fetchClassificationsAfterValueChange: () => {
+    fetchClassifications: () => {
       cy.wait(InterceptAliases.CaseClassification).logDetail(
         'should fetch new case classifications after the value changes'
```
Comment (SimonMilord) [inline packages/quantic/cypress/e2e/default-1/case-classification/case-classification-expectations.ts:169]: if you rename the expectation, do we want this detail to be updated as well then?

**Review by SimonMilord — APPROVED** [review]
Nice!

File: packages/quantic/cypress/e2e/default-1/case-classification/case-classification-expectations.ts (hunk 2 above)
Comment (mmitiche) [reply packages/quantic/cypress/e2e/default-1/case-classification/case-classification-expectations.ts:169]: sure!

Comment (mmitiche) [inline packages/quantic/cypress/e2e/default-1/case-classification/case-classification-expectations.ts:169]:
```suggestion
        'should fetch new case classifications'
```

**Pull request merged**